### PR TITLE
define model

### PR DESCRIPTION
**Pull request by s2snowflakes**

### CODE REVIEW

File: app/models/cart.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Cart < ApplicationRecord
+	belong_to :user
```
Comment (chuanhtuan) [inline app/models/cart.rb]: 2 space

File: app/models/category.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Category < ApplicationRecord
+  has_many :product
```
Comment (dieunb) [inline app/models/category.rb]: `has_many :products`

File: app/models/product.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Product < ApplicationRecord
+  belong_to :category
+  has_many :order_detail
```
Comment (dieunb) [inline app/models/product.rb]:
```
 +  has_many :order_details
 +  has_many :reviews
 +  has_many :images
 +  has_many :carts
```

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
```
Comment (dieunb) [inline app/models/user.rb]: `before_save {self.email = email.downcase}`

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+  validates :name,  presence: true, length: { maximum: 50 }
```
Comment (dieunb) [inline app/models/user.rb]:
```
 +  validates :name,  presence: true, length: {maximum: 50}
 +  validates :email, presence: true, length: {maximum: 255}
```

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+  validates :email, presence: true, length: { maximum: 255 }
+  has_many :order
```
Comment (dieunb) [inline app/models/user.rb]:
```
 +  has_many :orders
 +  has_many :reviews
 +  has_many :carts
```

**Pull request merged**